r/tests/testthat/test-dataset.R

```diff
@@ -383,15 +383,21 @@ test_that("Assembling a Dataset manually and getting a Table", {
   schm <- factory$Inspect()
   expect_is(schm, "Schema")
 
-  phys_schm <- ParquetFileReader$create(file.path(dataset_dir, 1, "file1.parquet"))$GetSchema()
+  file1 <- file.path(dataset_dir, 1, "file1.parquet")
+  file2 <- file.path(dataset_dir, 2, "file2.parquet")
+  phys_schm <- ParquetFileReader$create(file1)$GetSchema()
   expect_equal(names(phys_schm), names(df1))
   expect_equal(names(schm), c(names(phys_schm), "part"))
 
   src <- factory$Finish(schm)
-  expect_is(src, "Source")
+  expect_is(src, "FileSystemSource")
   expect_is(src$schema, "Schema")
+  expect_is(src$format, "ParquetFileFormat")
   expect_equal(names(schm), names(src$schema))
 
+
+  expect_equivalent(src$files, c(file1, file2))
```

Comment:
Is it guaranteed that files are returned in order?

Reply:
Yes: they will always be in a depth first visitation order.